File: PODM/SW/resource-manager/layer/src/main/java/com/intel/rsd/resourcemanager/layers/Response.java
/*
 * Copyright (c) 2017-2019 Intel Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.intel.rsd.resourcemanager.layers;

import com.fasterxml.jackson.databind.JsonNode;
import lombok.NonNull;
import lombok.ToString;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

@ToString
public class Response {
    private JsonNode body;

    private HttpHeaders httpHeaders;

    private HttpStatus httpStatus;

    public Response() {
    }

    public Response(@NonNull ResponseEntity<JsonNode> entity) {
        this(entity.getBody(), entity.getHeaders(), entity.getStatusCode());
    }

    public Response(JsonNode body, @NonNull HttpHeaders httpHeaders, @NonNull HttpStatus httpStatus) {
        this.body = body;
        this.httpHeaders = httpHeaders;
        this.httpStatus = httpStatus;
    }

    public Response(JsonNode body, @NonNull HttpStatus httpStatus) {
        this.body = body;
        this.httpHeaders = new HttpHeaders();
        this.httpStatus = httpStatus;
    }

    public JsonNode getBody() {
        return body;
    }

    public HttpHeaders getHttpHeaders() {
        return httpHeaders;
    }

    public HttpStatus getHttpStatus() {
        return httpStatus;
    }

    public ResponseEntity<JsonNode> toResponseEntity() {
        return new ResponseEntity<>(getBody(), getHttpHeaders(), getHttpStatus());
    }
}

File: PSME/application/src/rest/server/response.cpp
/*
 * Copyright (C) 2014 MediaSift Ltd.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
/*!
 * @copyright
 * Copyright (c) 2015-2019 Intel Corporation
 *
 * @copyright
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * @copyright
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * @copyright
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * */

#include "psme/rest/server/response.hpp"

using namespace psme::rest::server;

Response::Response()
	: m_status(status_2XX::OK)
{}


void Response::set_header(const std::string& header, const std::string& value) {
    m_headers[header] = value;
}

void Response::set_status(const std::uint32_t status_code) {
    m_status = status_code;
}

void Response::set_body(const std::string& body) {
    m_body = body;
}

Response& Response::operator<<(const std::string& rhs) {
    m_body += rhs;
    return (*this);
}

std::uint32_t Response::get_status() {
    return m_status;
}

std::size_t Response::get_body_size() {
    return m_body.size();
}

const std::string& Response::get_body() const {
    return m_body;
}

const Response::HeaderList& Response::get_headers() const {
    return m_headers;
}
